cmd/api/src/database/analysisrequest.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2024 Specter Ops, Inc.
+//
+// Licensed under the Apache License, Version 2.0
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package database
+
+import (
+	"context"
+	"database/sql"
+	"github.com/specterops/bloodhound/errors"
+	"github.com/specterops/bloodhound/log"
+	"github.com/specterops/bloodhound/src/model"
+	"time"
+)
+
+type AnalysisRequestData interface {
+	DeleteAnalysisRequest(ctx context.Context) error
+	GetAnalysisRequest(ctx context.Context) (model.AnalysisRequest, error)
+	HasAnalysisRequest(ctx context.Context) bool
+	HasDeletionRequest(ctx context.Context) bool
+	RequestAnalysis(ctx context.Context, requester string) error
+	RequestDeletion(ctx context.Context, requester string) error
```

Comment:
`RequestDeletion` is a little vague in that its not clear what exactly would be getting deleted.  other parts of the datapipe word it like `DeleteCollectedGraphData`, (see datapipe/delete.go).  lets keep it consistent with that?